cookbook/case_studies/feature_engineering/feast_integration/custom_provider/provider.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import pandas
+from feast.entity import Entity
+from feast.feature_table import FeatureTable
+from feast.feature_view import FeatureView
+from feast.infra.local import LocalProvider
+from feast.infra.offline_stores.file_source import FileSource
+from feast.infra.offline_stores.offline_store import RetrievalJob
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import RepoConfig
+from flytekit.core.context_manager import FlyteContext
+from tqdm import tqdm
+
+
+class FlyteCustomProvider(LocalProvider):
+    def __init__(self, config: RepoConfig, repo_path):
+        super().__init__(config)
+
+    def materialize_single_feature_view(
+        self,
+        config: RepoConfig,
+        feature_view: FeatureView,
+        start_date: datetime,
+        end_date: datetime,
+        registry: Registry,
+        project: str,
+        tqdm_builder: Callable[[int], tqdm],
+    ) -> None:
+        # materialize_single_feature_view loads the latest feature values for a specific feature value from the offline
+        # store into the online store.
+        # This method can be overridden to also launch custom batch ingestion jobs that loads the latest batch feature
+        # values into the online store.
+        #
+
+        self._localize_feature_view(feature_view)
+
+        # Replace the line below with your custom logic in order to launch your own batch ingestion job
+        super().materialize_single_feature_view(
+            config, feature_view, start_date, end_date, registry, project, tqdm_builder
+        )
+        print("Launching custom batch jobs is pretty easy...")
```

Comment:
i dont follow, will this work in Flyte Task?

Reply:
This code is meant to be overridden.

Reply:
oops, this print statement is not supposed to be there. This is from the [example](https://github.com/feast-dev/feast-custom-provider-demo/blob/master/feast_custom_provider/custom_provider.py#L90) custom provider.